agent-bridge-datastore/src/main/java/com/newrelic/agent/bridge/datastore/ExpiringValueCache.java

```diff
@@ -0,0 +1,108 @@
+package com.newrelic.agent.bridge.datastore;
+
+import com.newrelic.agent.bridge.AgentBridge;
+
+import java.time.Instant;
+import java.util.AbstractMap;
+import java.util.Set;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+
+/**
+ * A cache implementation that uses a composition pattern to wrap a {@link ConcurrentHashMap}
+ * populated with {@link TimestampedMapValue} instances. An instance of this class is constructed
+ * with function that contains the logic of when to evict items from the cache.
+ *
+ * @param <K> the class type of the cache key
+ * @param <V> the class type of the value object
+ */
+public class ExpiringValueCache<K, V> {
```

Comment:
This agent uses [Caffeie](https://github.com/ben-manes/caffeine/) in [several places](https://github.com/newrelic/newrelic-java-agent/search?q=caffeine); is there a reason not to use it here as well? Using it also avoids the need to start an extra thread.

Reply:
Yes, we tried using Caffeine here, but due to the way this module is loaded, we were not able to use it.